frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCompareWithParentAction.java

```diff
@@ -0,0 +1,68 @@
+package com.virtuslab.gitmachete.frontend.actions.base;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.fmt;
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getNonHtmlString;
+
+import java.util.Collections;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.fileEditor.FileDocumentManager;
+import git4idea.branch.GitBrancher;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+public abstract class BaseCompareWithParentAction extends BaseGitMacheteRepositoryReadyAction implements IBranchNameProvider {
+
+  @Override
+  @UIEffect
+  protected void onUpdate(AnActionEvent anActionEvent) {
+    super.onUpdate(anActionEvent);
+
+    val presentation = anActionEvent.getPresentation();
+    if (!presentation.isEnabled()) {
+      return;
+    }
+
+    val branchName = getNameOfBranchUnderAction(anActionEvent);
+    val gitRepository = getSelectedGitRepository(anActionEvent);
+    val managedBranch = getManagedBranchByName(anActionEvent, branchName);
+
+    if (gitRepository == null || managedBranch == null) {
+      return;
+    }
+
+    if (managedBranch.isRoot()) {
+      presentation.setEnabled(false);
+      presentation.setDescription(
+          fmt(getNonHtmlString("action.GitMachete.BaseCompareWithParentAction.description.disabled.branch-is-root"),
+              managedBranch.getName()));
+      return;
+    }
+
+    val parent = managedBranch.asNonRoot().getParent();
+
+    String description = fmt(getNonHtmlString("action.GitMachete.BaseCompareWithParentAction.description.precise"),
+        parent.getName(),
+        managedBranch.getName());
+    presentation.setDescription(description);
+  }
+
+  @Override
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    FileDocumentManager.getInstance().saveAllDocuments();
+
+    val project = getProject(anActionEvent);
+    val branchName = getNameOfBranchUnderAction(anActionEvent);
+    val gitRepository = getSelectedGitRepository(anActionEvent);
+    val managedBranch = getManagedBranchByName(anActionEvent, branchName);
+
+    if (gitRepository == null || managedBranch == null || managedBranch.isRoot()) {
+      return;
+    }
+
+    val parent = managedBranch.asNonRoot().getParent();
+    GitBrancher.getInstance(project).compare(parent.getName(), Collections.singletonList(gitRepository));
```

Comment:
Hmmm  does it compare selected branch or rather the current working directory? 🤔 given that the other branch is not specified (i.e. it's not `compareAny` that's called)

Reply:
Have you tried it? 

It compares branches based on commits. Working directory changes are ignored. Here is an example:
<img width="901" alt="image" src="https://user-images.githubusercontent.com/19799111/218021578-627ef94c-2d4d-4962-944c-f38912cc7dfb.png">

Reply:
> Working directory changes are ignored. Here is an example:

Whoops indeed 😓 anyway, another issue:

Hmmm I'm checking the current commit... and getting the following unexpected result on each child of `foo-allow-ownership-link` which is the current branch:

<img width="1389" alt="image" src="https://user-images.githubusercontent.com/3383210/218317758-ab55c4ac-7917-4eab-9f6c-a0245f265370.png">

The only branch for which I'm getting the expected result is `foo-allow-ownership-link`:

<img width="1394" alt="image" src="https://user-images.githubusercontent.com/3383210/218317791-199a1e30-2db5-41bc-8b97-041272a4e443.png">

Sooo... it looks that the selected branch is ignored (?)

Reply:
oh, ofc. I must have been drunk or smth ... 🤦 `compareAny` ofc

Reply:
pls try it now